staging/src/k8s.io/apimachinery/pkg/util/diff/diff_test.go

```diff
@@ -27,3 +27,26 @@ func TestStringDiff(t *testing.T) {
 		t.Errorf("diff returned %v", diff)
 	}
 }
+
+func TestLegacyDiff(t *testing.T) {
+	equalResult := legacyDiff(true, true)
+	diffResult := legacyDiff(true, "string")
+
+	if equalResult != "" {
+		t.Errorf("two param type are same, result should be empty, but now the result is: %v", equalResult)
+	}
+
+	expectDiffResult := "  interface{}(\n- \tbool(true),\n+ \tstring(\"string\"),\n  )\n"
+	if diffResult != expectDiffResult {
+		t.Errorf("two param type are not same, expect result is: %v, but now the result is: %v", expectDiffResult, diffResult)
+	}
+}
+
+func TestObjectGoPrintSideBySide(t *testing.T) {
+	result := ObjectGoPrintSideBySide("a", "ab")
```

Comment:
Can you use a more complex test case that can show the power of the `ObjectGoPrintSideBySide` function, e.g., subtle diffs made apparent by `ObjectGoPrintSideBySide`?